### ISSUE TITLE
[Enhancement] Make Ice Form Under Blocks Even If A Block Is Above It

### ISSUE THREAD

**Issue by jredfox** [TheWeatherPony/TheSeasonsMod#43]
Version: beta8c4 , untested on latest
Priority: High
Images:  http://imgur.com/p23heWF

**Problem:**
Ice doesn't form in water unless there is no block above it. Now you should understand if you fix this ice will start forming in caves even by lava. 

**Possible Solution:**
I suggest making it form only on the surface or on a certion y limit and would be configurable per biome and dimension.

**Comment by PitchBright** [comment]
Just thinking out loud, but I don't think it's a bad thing if water freezes below ground. If it's cold enough to freeze, it's cold enough to freeze, doesn't matter where it is.

**Comment by jredfox** [comment]
yeah but, in caves in real life they usually have flowing water except vanilla minecraft doesn't do that also in real life I am unsure if seasonal biomes caves freeze. I know they don't if the cave has lots of lava.  I guess I could add a suggestion for freezing caves at freezing biomes and outside cave freezes except that is basically the elemental caves mod that did a great job at it.  going to be gone for hours I will respond asap when I get back.

**Comment by PitchBright** [comment]
Water doesn't care what "season" it is. Water freezes when it reaches 0 degrees centigrade… wherever it is.

This mod isn't about adding biomes. You gotta drop that whole train of thought. Get a biome mod that adds biomes if you want new biomes. Get a temperature mod that lets you control specific block temperatures if you wanna control temperatures of blocks.
